### ISSUE TITLE
PanelMenu: Click event not triggered by keyboard interaction for header items with custom templates

### ISSUE THREAD

**Issue by sane6** [openvi-foundation/openvue#283]
### Describe the bug

This bug report references #6387.
When using the PanelMenu component with a custom template for elements via a slot, the click event is not triggered when using keyboard interactions, such as pressing the Enter or Space key, for root menu items. It does work with a mouse click event.

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-7pd1syyv?file=src%2FApp.vue

### Environment

StackBlitz, as well as VitePress

### Vue version

3.5.3

### PrimeVue version

4.3.3

### Steps to reproduce the behavior

1. Go to this link to replicate the issue: https://stackblitz.com/edit/primevue-4-vite-issue-template-phvq15?file=src%2FApp.vue%3AL7
2. Using the keyboard, focus on the first item (Test) and press Enter or Space.
3. Continue using the keyboard to navigate and open "Mail" -> "Compose". (In this case, the click event is fired even though the keyboard is used to navigate through the menu.)

Using the suggested fix in #6387, adding `tabIndex` to the anchor tag, did not solve the issue (see StackBlitz reproduction).

### Expected behavior

The expected behavior is that the same click event should fire when interacting with top-level items using either the mouse or the keyboard, just as it does for item links / submenu items.

Copied to track in our fork; not affiliated with the original report being closed/resolved upstream

Originally reported at: [primefaces#7523](https://github.com/primefaces/primevue/issues/7523)
Originally reported by: https://github.com/MasterCarl
